jib-core/src/main/java/com/google/cloud/tools/jib/ncache/DefaultCacheStorageWriter.java

```diff
@@ -154,30 +220,60 @@ CacheEntry write(CacheWrite cacheWrite) throws IOException {
               .setLayerDiffId(writtenLayer.layerDiffId)
               .setLayerSize(writtenLayer.layerSize)
               .setLayerBlob(Blobs.from(layerFile));
-      if (cacheWrite.getMetadataBlob().isPresent()) {
+      if (uncompressedCacheWrite.getMetadataBlob().isPresent()) {
         Path metadataFile = defaultCacheStorageFiles.getMetadataFile(writtenLayer.layerDigest);
         cacheEntryBuilder.setMetadataBlob(Blobs.from(metadataFile)).build();
       }
 
       // Write the selector file.
-      if (cacheWrite.getSelector().isPresent()) {
-        writeSelector(cacheWrite.getSelector().get(), writtenLayer.layerDigest);
+      if (uncompressedCacheWrite.getSelector().isPresent()) {
+        writeSelector(uncompressedCacheWrite.getSelector().get(), writtenLayer.layerDigest);
       }
 
       return cacheEntryBuilder.build();
     }
   }
 
   /**
-   * Writes the {@code layerBlob} to the {@code layerDirectory}.
+   * Writes a compressed {@code layerBlob} to the {@code layerDirectory}.
    *
-   * @param layerBlob the layer {@link Blob}
+   * @param compressedLayerBlob the compressed layer {@link Blob}
    * @param layerDirectory the directory for the layer
    * @return a {@link WrittenLayer} with the written layer information
    * @throws IOException if an I/O exception occurs
    */
-  private WrittenLayer writeLayerBlobToDirectory(Blob layerBlob, Path layerDirectory)
-      throws IOException {
+  private WrittenLayer writeCompressedLayerBlobToDirectory(
+      Blob compressedLayerBlob, Path layerDirectory) throws IOException {
+    // Writes the layer file to the temporary directory.
+    Path temporaryLayerFile = Files.createTempFile(layerDirectory, null, null);
```

Comment:
Just saying: we don't really need to create a file, since we'll just overwrite it. We just need a random name, but admittedly, `Files.createTempFile()` is an easy way to have a random name.

Reply:
Yea, maybe we can just use a static file name like "layer".

Reply:
I'm going to actually create a `TemporaryFile` class to encapsulate this logic better.

Reply:
Actually, nevermind, I'll leave that to a separate PR.